canonical-web-and-design/maas-ui
# Add save functionality to Storage form + build ActionButton

## Done
- Added save functionality to Storage form
- Reworked Storage form to use FormikField component
- Updated Storage form styling (no title, left-aligned button)
- Added ActionButton component which wraps Button. It takes props to determine whether it's in a loading or success state. It takes an explicit width prop which is a bit crap but I couldn't figure out how to store button width in the hooks world and figured this works fine for now :man_shrugging: 
- Added some hooks that essentially "fake" a delay after clicking Save, even though saving happens almost instantly and therefore saving state is `false`. Success state is simply determined as "saving state moved from true => false" so will probs need some handling for when that happens due to an error.

## QA
- Go to Storage and check that the button is disabled
- Change some values and check that button is no longer disabled
- Change values back to original (without saving) and check that button becomes disabled again
- Save some changes and check that the button text is replaced by a spinner, and then after a short time it changes to a tick
- Refresh the page and check that state has persisted

## Screenshots
### No values changed
![Screen Shot 2019-08-12 at 18 51 44-fullpage](https://user-images.githubusercontent.com/25733845/62882580-8510dc80-bd32-11e9-8e1e-a691d99c6d04.png)


### Values changed
![Screen Shot 2019-08-12 at 18 51 54-fullpage](https://user-images.githubusercontent.com/25733845/62882589-880bcd00-bd32-11e9-8622-f0153f8657bd.png)


### Saving
![Screen Shot 2019-08-12 at 18 52 00-fullpage](https://user-images.githubusercontent.com/25733845/62882596-8b06bd80-bd32-11e9-8ea4-ba81fc733220.png)


### Saved
![Screen Shot 2019-08-12 at 18 52 01-fullpage](https://user-images.githubusercontent.com/25733845/62882600-8e01ae00-bd32-11e9-97d5-e3176af7c88c.png)
